src/hash.h

```diff
@@ -111,23 +111,38 @@ class CHashWriter {
     int GetVersion() const { return nVersion; }
 
     void write(const char *pch, size_t size) {
-        ctx.Write({(const uint8_t *)pch, size});
+        ctx.Write((const uint8_t *)pch, size);
     }
 
-    // invalidates the object
+    /** Compute the double-SHA256 hash of all data written to this object.
```

Comment:
How is OP_CDS doing the single round of SHA256 right now?

Reply:
```
                        if (vchSig.size()) {
                            valtype vchHash(32);
                            CSHA256()
                                .Write(vchMessage.data(), vchMessage.size())
                                .Finalize(vchHash.data());
                            fSuccess = checker.VerifySignature(
                                vchSig, CPubKey(vchPubKey), uint256(vchHash));
                            metrics.nSigChecks += 1;
```

May want to refactor this as you're doing this. IDK.

Reply:
Great idea.